docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+- Each message contains a list of hashes of every previous messages that the signer has perceived, **forming a chain**.
+- Like PoW, the longest chat chain is considered a candidate for the canonical one.
+- The consensus leader (block proposer) for the round plays a special role.
+  - the leader can broadcast a chat as well, but such chat is considered a **semifinalized point**.
+  - A *faithful* leader would **frequently** insert its `ack` chat **on the (seemingly) longest chain**.
+  - A *good* leader must not commit a fork (semifinalizing two conflicting chains)
+- On top of the recent leader-semifinalized chat, other members continue chatting using the longest-chain rule until the next finalization point.
+- Once the governance reaches an eligible agenda, the leader must semifinalize the chat chain for the last time, and include it in the block. If the block is finalized by the consensus then the chat log for the block is finalized as well.
+- A *good* leader must include the last-semifinalized chat chain which is the one that the other members would consider the canonical one.
+
+### Why doesn't the leader just serve a chat server?
+
+The reason that the leader only 'semifinalizes' the chat chain, instead of running a full chat server, is that it is more censorship-resistant.
+
+- Compared to a full chat server, the longest-chain protocol has a
+  **weak** (because of the lack of network synchrony) **consensus** (longest chain) of the canonical total chat ordering, among the members.
+- Thus if the leader attempts censorship (i.e., repeatedly semifinalizes a chat chain that is not the longest one observed by other members), it's easy to *notice* that, though it's not theoretically verifiable due to the assumption of network asynchrony. (it might be a coincidence originated from the possible network delay around the leader)
+- Nonetheless, if the leader attempts censorship more, it becomes more suspicious.
+
+Note that if there is a malicious member who tries to spam the chat, the leader will not finalize the spam chain, and the other members would recognize such censorship as something `not bad`.
+
+### What happens in the consensus?
+
+The consensus leader (*block proposer*; don't be confused with the *agenda proposer*) includes an eligible agenda (if there were multiple eligible agendas, the leader just chooses one) in the block, and proposes it.
+**A valid block MUST contain an eligible agenda.**. In other words, if there is no eligible agenda, there is no block progress in the blockchain.
+
+In addition to the agenda, the leader can also include some other transactions.
+
+1. RecordChatLog (chatting logs for the very height): this doesn't change the state, but is used as the governance minutes for the agenda.
+2. ReportMisbehavior (double votes in the consensus and chat finalization misbehaviors by a past consensus leader): this will lead to the slashing of the voting power of the misbehaving validator.
```

Comment:
Then how the voting power be slashed? Is it automated, or does it require proposing an agenda?

Reply:
Since it's not the part of an agenda, it does not require a governance approval.

Reply:
This is a common design in almost every stake-based chains, so looks quite clear to me. I will put a word 'automatically' just in case.

Reply:
okay